### PR TITLE
Practical-8

### PR DESCRIPTION
**Pull request by desaianiruddh**

### CODE REVIEW

File: user-management-react-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+function App() {
+  const navigate = useNavigate();
+  const isUserLoggedIn = useSelector((state) => state.userData.isUserLoggedIn);
+  useEffect(() => {
+    isUserLoggedIn ? navigate('/home') : navigate('/');
+  }, [isUserLoggedIn, navigate]);
+  return (
+    <div className="app">
+      <Routes>
+        <Route path="/" element={<SignUp />} />
```
Comment (Rutvi25) [inline user-management-react-app/src/App.js:19]: path name should be "signup" for signup page

File: user-management-react-app/src/components/UserCard/UserCard.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+const UserCard = () => {
+  const userData = useSelector((state) => state.userData.userLoginData);
+  console.log(userData);
```
Comment (Rutvi25) [inline user-management-react-app/src/components/UserCard/UserCard.jsx:6]: remove logs

File: user-management-react-app/src/reducers/userData.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const intialData = {
+  isUserLoggedIn: false,
+  userLoginData: {},
+};
+
+const userData = (state = intialData, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case 'SUBMIT_FORM': {
```
Comment (Rutvi25) [inline user-management-react-app/src/reducers/userData.js]: no need of {},
you can directly write return statement
case 'SUBMIT_FORM':
  return {
  ...
  };

File: user-management-react-app/src/App.js (hunk 1 above)
Comment (Rutvi25) [inline user-management-react-app/src/App.js]: path name should be "signup" for signup page

File: user-management-react-app/src/components/UserCard/UserCard.jsx (hunk 2 above)
Comment (Rutvi25) [inline user-management-react-app/src/components/UserCard/UserCard.jsx]: remove logs

**Review by Rutvi25 — CHANGES_REQUESTED** [review]

**Review by Rutvi25 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: user-management-react-app/src/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export const submitForm = (values) => {
+  return {
+    type: 'SUBMIT_FORM',
+    payload: {
+      ...values,
+      profilePhoto: URL.createObjectURL(values.profilePhoto),
```
Comment (sheth-jay) [inline user-management-react-app/src/actions/index.js]: This should be done inside OnSubmit function.

File: user-management-react-app/src/actions/index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export const submitForm = (values) => {
+  return {
+    type: 'SUBMIT_FORM',
```
Comment (sheth-jay) [inline user-management-react-app/src/actions/index.js]: Change action name to something more meaningful name. As in future you may have multiple forms at that time SUBMIT_FORM will be confusing.

File: user-management-react-app/src/components/Pages/SignUp/SignUp.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+import SignUpForm from '../../SignUpForm/SignUpForm';
+import signUpImg from '../../../img/signup.png';
+import './signUp.css';
+
+const SignUp = () => {
+  return (
+    <div
+      className={`bg-white mt-4 ${
+        window.innerWidth > 1048
```
Comment (sheth-jay) [inline user-management-react-app/src/components/Pages/SignUp/SignUp.jsx]: Always use appropriate break points.

**Review by sheth-jay — CHANGES_REQUESTED** [review]

**Review by sheth-jay — APPROVED** [review]

**Pull request merged**